lib/ecto/adapters/postgres/connection.ex

```diff
@@ -682,6 +682,13 @@ if Code.ensure_loaded?(Postgrex) do
       "ALTER TABLE #{quote_table(constraint.prefix, constraint.table)} DROP CONSTRAINT #{quote_name(constraint.name)}"
     end
 
+    def execute_ddl({:create, %View{} = view}) do
+      {query, params, _key} = Ecto.Query.Planner.prepare(view.query, :all, Ecto.Adapters.Postgres, 0)
```

Comment:
Yeah, also wanted to that from there initially, but application repo isn't available there either. The other point here is that repo is used by `Ecto.Adapters.SQL.to_sql` is when the query is cached, which isn't really a use case for migrations, so maybe we can try to simplify the `to_sql` API.

Aside from that the only place in the stack that has repo available, is the `Ecto.Adapters.SQL.execute_ddl`.